homeassistant/components/media_player/yamaha.py

```diff
@@ -110,10 +149,16 @@ def __init__(self, name, receiver, source_ignore, source_names):
         self._name = name
         self._zone = receiver.zone
 
+    def __eq__(self, other):
```

Comment:
Cool! One question though. Since all added devices are globally stored in the config (so they can be iterated by the service handler), I basically want to do ``hass.data[DATA_YAMAHA].append(device)`` only if a devices is actually added. How would you recommend solving that issue?